controllers/nodehealthcheck_controller.go

```diff
@@ -479,6 +479,13 @@ func (r *NodeHealthCheckReconciler) remediate(node *v1.Node, nhc *remediationv1a
 	// always update status, in case patching it failed during last reconcile
 	resources.UpdateStatusRemediationStarted(node, nhc, remediationCR)
 
+	// Adding a watch for 2 usecases:
+	// - watch for conditions (e.g. "Succeeded" for escalating remediations)
+	// - watch for removed finalizers / CR deletion, for potentially start the next control plane node remediation
+	if err = r.addWatch(remediationCR); err != nil {
```

Comment:
Nit: Any reason not to push this higher ? (i.e immediately after `remediationCR` is generated successfully)

Reply:
- restarted NHC pods didn't watch CRs
- NHC didn't watch CRs for "classic" remediation